packages/puppeteer-core/src/bidi/HTTPResponse.ts

```
@@ -7,86 +7,86 @@ import type * as Bidi from 'chromium-bidi/lib/cjs/protocol/protocol.js';
-    // TODO: File and issue with BiDi spec
-    this.#timings = null;
-
-    // TODO: Removed once the Firefox implementation is compliant with https://w3c.github.io/webdriver-bidi/#get-the-response-data.
-    for (const header of response.headers || []) {
-      // TODO: How to handle Binary Headers
-      // https://w3c.github.io/webdriver-bidi/#type-network-Header
-      if (header.value.type === 'string') {
-        this.#headers[header.name.toLowerCase()] = header.value.value;
-      }
-    }
+  #initialize() {
+    this.#request.frame()?.page().emit(PageEvent.Response, this);
   }
 
+  @invokeAtMostOnceForArguments
   override remoteAddress(): RemoteAddress {
-    return this.#remoteAddress;
+    return {
+      ip: '',
+      port: -1,
+    };
   }
 
   override url(): string {
-    return this.#url;
+    return this.#data.url;
   }
 
   override status(): number {
-    return this.#status;
+    return this.#data.status;
   }
 
   override statusText(): string {
-    return this.#statusText;
+    return this.#data.statusText;
   }
 
   override headers(): Record<string, string> {
-    return this.#headers;
+    const headers: Record<string, string> = {};
+    // TODO: Remove once the Firefox implementation is compliant with https://w3c.github.io/webdriver-bidi/#get-the-response-data.
+    for (const header of this.#data.headers || []) {
+      // TODO: How to handle Binary Headers
+      // https://w3c.github.io/webdriver-bidi/#type-network-Header
+      if (header.value.type === 'string') {
+        headers[header.name.toLowerCase()] = header.value.value;
+      }
+    }
+    return headers;
   }
 
   override request(): BidiHTTPRequest {
     return this.#request;
   }
 
   override fromCache(): boolean {
-    return this.#fromCache;
+    return this.#data.fromCache;
   }
 
   override timing(): Protocol.Network.ResourceTiming | null {
-    return this.#timings as any;
+    // TODO: File and issue with BiDi spec
+    return null;
```

let's throw UnsupportedOperation